src/tr1/game/camera.c

```diff
@@ -302,115 +338,56 @@ static void M_UpdatePhotoMode(void)
         }
-        SHIFT_POS(g_Camera.pos, shift);
-        SHIFT_POS(g_Camera.target, shift);
-    }
-
-    if (dir_shift_input) {
-        const int32_t distance = CAM_SPEED_SHIFT((WALL_L * 5.0) / LOGIC_FPS);
-
-        PHD_ANGLE angle = g_Camera.target_angle;
-        PHD_ANGLE elevation = g_Camera.target_elevation;
-
-        if (g_Input.camera_forward || g_Input.camera_back) {
-            int32_t dy = 0;
-            if (g_Input.camera_forward) {
-                dy = 1;
-            } else {
-                angle += PHD_180;
-                dy = -1;
-            }
-
-            const XYZ_32 shift = {
-                .x = SHIFT_X(distance, elevation, angle),
-                .z = SHIFT_Z(distance, elevation, angle),
-                .y = SHIFT_Y(dy, distance, elevation),
-            };
-            SHIFT_POS(g_Camera.pos, shift);
-            SHIFT_POS(g_Camera.target, shift);
-
-            Math_GetVectorAngles(
-                g_Camera.target.x - g_Camera.pos.x,
-                g_Camera.target.y - g_Camera.pos.y,
-                g_Camera.target.z - g_Camera.pos.z, angles);
-            angle = angles[0];
-            elevation = angles[1];
+        if (g_Input.camera_forward) {
+            M_PhotoModeShift(
+                g_Camera.target_angle, g_Camera.target_elevation, -1);
+        } else if (g_Input.camera_back) {
+            M_PhotoModeShift(
+                g_Camera.target_angle + PHD_180, g_Camera.target_elevation, 1);
         }
-
-        if (g_Input.camera_left || g_Input.camera_right) {
-            if (g_Input.camera_left) {
-                angle -= PHD_90;
-            } else {
-                angle += PHD_90;
-            }
-
-            const XYZ_32 shift = {
-                .x = SHIFT_X(distance, elevation, angle),
-                .z = SHIFT_Z(distance, elevation, angle),
-                .y = 0,
-            };
-            SHIFT_POS(g_Camera.pos, shift);
-            SHIFT_POS(g_Camera.target, shift);
+        if (g_Input.camera_up) {
+            M_PhotoModeShift(
+                g_Camera.target_angle, g_Camera.target_elevation + PHD_90, -1);
+        } else if (g_Input.camera_down) {
+            M_PhotoModeShift(
+                g_Camera.target_angle, g_Camera.target_elevation - PHD_90, -1);
         }
     }
 
     if (rot_input) {
         if (g_Input.forward) {
-            g_Camera.target_elevation -= CAM_ROT_SHIFT;
+            M_PhotoModeRotate(
+                g_Camera.target_angle,
+                g_Camera.target_elevation - CAM_ROT_SHIFT);
         } else if (g_Input.back) {
-            g_Camera.target_elevation += CAM_ROT_SHIFT;
+            M_PhotoModeRotate(
+                g_Camera.target_angle,
+                g_Camera.target_elevation + CAM_ROT_SHIFT);
         }
-        CLAMP(
-            g_Camera.target_elevation, -PHOTO_MAX_PITCH_ROLL,
-            PHOTO_MAX_PITCH_ROLL);
 
         if (g_Input.left) {
-            g_Camera.target_angle -= (int16_t)CAM_ROT_SHIFT;
```

Comment:
@rr- Re @aredfan's finding, this was also an issue I had in the original build, but casting to `int16_t` resolved it then. Perhaps there is a better approach though.
Apologies, I missed this one.